src/ASM/FiniteElement.C

```diff
@@ -42,7 +48,7 @@ std::ostream& FiniteElement::write (std::ostream& os) const
 std::ostream& MxFiniteElement::write (std::ostream& os) const
 {
   this->FiniteElement::write(os);
-  for (size_t b = 0; b+1 < Nx.size(); b++)
+  for (size_t b = 0; b < Nx.size(); b++)
   {
```

Comment:
OK, just disregard my previous comment, your suggested fix is correct. (I did not see the rest of the body, which you need to do in order to see that)